src/dvc_studio_client/post_live_metrics.py

```diff
@@ -25,7 +25,11 @@ def _get_remote_url() -> str:
     from dulwich.repo import Repo
 
     with Repo.discover() as repo:
-        _remote, url = get_remote_repo(repo)
+        try:
+            _remote, url = get_remote_repo(repo)
+        except IndexError:
```

Comment:
I think this should actually be the default behavior in https://github.com/jelmer/dulwich/blob/1ad7903cd8c5410dc6573ea27d284a4fa79ec184/dulwich/porcelain.py#L1609 (returning `origin` on detached HEAD) as it is what git is doing in `git ls-remote --get-url`.

Merging this here as a quick fix